core/src/main/java/nu/yona/server/device/service/DeviceService.java

```diff
@@ -270,13 +283,18 @@ public void deleteDevice(UUID userId, UUID deviceId)
 			UUID deviceAnonymizedId = deviceEntity.getDeviceAnonymized().getId();
 
 			deleteDeviceWithoutBuddyNotification(userEntity, deviceEntity);
-			userAnonymizedService.updateUserAnonymized(userEntity.getAnonymized().getId());
+			evictUserAnonymizedFromCache(userEntity);
 
 			sendDeviceChangeMessageToBuddies(DeviceChange.DELETE, userEntity, Optional.of(oldName), Optional.empty(),
 					deviceAnonymizedId);
 		});
 	}
 
+	private UserAnonymizedDto evictUserAnonymizedFromCache(User userEntity)
```

Comment:
And call that instead of ``evictUserAnonymizedFromCache``? I created this method to make explicit that we're evicting the user anonymized from the cache. That wouldn't be clear anymore of we would be calling an update method on the ``UserAnonymizedService``.